File: src/main/java/net/benwoodworth/fastcraft/core/api/dependencies/recipes/FcShapedRecipe.java
package net.benwoodworth.fastcraft.core.api.dependencies.recipes;

import net.benwoodworth.fastcraft.core.api.dependencies.inventory.FcItem;
import net.benwoodworth.fastcraft.core.api.dependencies.inventory.ItemGrid;
import org.jetbrains.annotations.NotNull;

/**
 * A shaped recipe.
 *
 * @param <TItem> The native item type.
 */
public abstract class FcShapedRecipe<TItem> extends FcRecipe<TItem> {

    /**
     * Get the recipe's shape.
     *
     * @return Returns the recipe's shape.
     */
    @NotNull
    public abstract ItemGrid<FcItem<TItem>> getRecipeShape();
}

File: src/main/kotlin/net/benwoodworth/fastcraft/core/api/dependencies/recipes/FcShapedRecipe.kt
package net.benwoodworth.fastcraft.core.api.dependencies.recipes

import net.benwoodworth.fastcraft.core.api.dependencies.inventory.FcItem
import net.benwoodworth.fastcraft.core.api.dependencies.inventory.ItemGrid

/**
 * A shaped recipe.
 *
 * @param TItem the native item type
 */
abstract class FcShapedRecipe<TItem> : FcRecipe<TItem>() {

    /** The recipe's shape. */
    abstract val recipeShape: ItemGrid<FcItem<TItem>>
}
